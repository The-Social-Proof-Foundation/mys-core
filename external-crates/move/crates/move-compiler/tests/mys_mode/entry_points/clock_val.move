// invalid, Clock by value

module a::m {
    public entry fun no_clock_val(_: mys::clock::Clock) {
        abort 0
    }
}

module mys::clock {
    struct Clock has key {
        id: mys::object::UID,
    }
}

module mys::object {
    struct UID has store {
        id: address,
    }
}
