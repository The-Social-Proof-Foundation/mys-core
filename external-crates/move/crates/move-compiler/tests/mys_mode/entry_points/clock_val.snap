---
source: crates/move-compiler/tests/move_check_testsuite.rs
info:
  flavor: mys
  edition: legacy
  lint: false
---
error[Mys E02002]: invalid 'entry' function signature
  ┌─ tests/mys_mode/entry_points/clock_val.move:4:35
  │
4 │     public entry fun no_clock_val(_: mys::clock::Clock) {
  │            -----                  ^  ----------------- mys::clock::Clock must be passed by immutable reference, e.g. '&mys::clock::Clock'
  │            │                      │   
  │            │                      Invalid 'entry' parameter type for parameter '_'
  │            'no_clock_val' was declared 'entry' here
